docs/csharp/language-reference/compiler-messages/cs0518.md

```diff
@@ -11,8 +11,11 @@ ms.assetid: b0b61cbb-c9a7-48c9-9e60-7cdd5ecb3e6c
 # Compiler Error CS0518
 
 Predefined type 'type' is not defined or imported  
-  
-The main cause for this problem is that the project is not importing mscorlib.dll, which defines the entire System namespace. This can be caused by one of the following:  
+
+> [!NOTE]
+> The resolution for this error depends on whether you're using a modern SDK-style project (`.csproj` files that start with `<Project Sdk="Microsoft.NET.Sdk">`) or legacy project formats. SDK-style projects manage runtime references automatically through the `<TargetFramework>` property.
+
+The main cause for this problem is that the project cannot access the predefined types from the .NET runtime library. In modern SDK-style projects, this is typically due to an incorrect or missing `<TargetFramework>` specification. In legacy projects, this issue is caused by not importing mscorlib.dll, which defines the entire System namespace. This can be caused by one of the following:  
```

Comment:
```suggestion
The main cause for this problem is that the project cannot access the predefined types from the .NET runtime library. In modern SDK-style projects, this is typically due to an incorrect or missing `<TargetFramework>` specification. In legacy projects, this issue is caused by not importing mscorlib.dll, which defines the entire <xref:System> namespace. This can be caused by one of the following:  
```